api/graphql/schema.graphql

```diff
@@ -165,10 +162,189 @@ input UpdateUserInput {
     preferredLang: Language @rename(attribute: "preferred_lang")
 }
 
+input LocalizedStringInput {
+    en: String
+    fr: String
+}
+
+input CreateUserBelongsTo {
+    connect: ID
+}
+
+input CreatePoolBelongsTo {
+    connect: ID
+}
+
+input CreateClassificationInput {
+    name: LocalizedStringInput
+    group: String!
+    level: Int!
+    minSalary: Int @rename(attribute: "min_salary")
+    maxSalary: Int @rename(attribute: "max_salary")
+}
+
+input CreateClassificationBelongsToMany {
+    create: [CreateClassificationInput!]
+    connect: [ID!]
+}
+
+input CreateCmoAssetInput {
+    key: String!
+    name: LocalizedStringInput!
+    description: LocalizedStringInput
+}
+
+input CreateCmoAssetBelongsToMany {
+    create: [CreateCmoAssetInput!]
+    connect: [ID!]
+}
+
+input CreateOperationalRequirementInput {
+    key: String
+    name: LocalizedStringInput!
+    description: LocalizedStringInput
+}
+
+input CreateOperationalRequirementBelongsToMany {
+    create: [CreateOperationalRequirementInput!]
+    connect: [ID!]
+}
+
+input CreatePoolCandidateInput {
+    pool: CreatePoolBelongsTo @belongsTo
+    user: CreateUserBelongsTo @belongsTo
+    cmoIdentifier: ID @rename(attribute: "cmo_identifier")
+    expiryDate: Date
+    isWoman: Boolean @rename(attribute: "is_woman")
+    hasDisability: Boolean @rename(attribute: "has_disability")
+    isIndigenous: Boolean @rename(attribute: "is_indigenous")
+    isVisibleMinority: Boolean @rename(attribute: "is_visible_minority")
+    hasDiploma: Boolean @rename(attribute: "has_diploma")
+    languageAbility: LanguageAbility @rename(attribute: "language_ability")
+    locationPreferences: [WorkRegion] @rename(attribute: "location_preferences")
+    acceptedOperationalRequirements: [OperationalRequirement] @belongsToMany
+    expectedSalary: [SalaryRange] @rename(attribute: "expected_salary")
+    expectedClassifications: [Classification] @belongsToMany
+    cmoAssets: [CmoAsset] @belongsToMany
+    status: PoolCandidateStatus
+}
+
+input CreatePoolCandidateHasMany {
+    create: [CreatePoolCandidateInput!]
+}
+
+input CreatePoolInput {
+    owner: CreateUserBelongsTo @rename(attribute: "user")
+    name: LocalizedStringInput
+    description: LocalizedStringInput
+    classifications: CreateClassificationBelongsToMany
+    assetCriteria: CreateCmoAssetBelongsToMany
+    essentialCriteria: CreateCmoAssetBelongsToMany
+    operationalRequirements: CreateOperationalRequirementBelongsToMany
+    poolCandidates: CreatePoolCandidateHasMany
+}
+
+input UpdateUserBelongsTo {
+    connect: ID
+}
+
+input UpdatePoolBelongsTo {
+    connect: ID
+}
+
+input UpdateClassificationInput {
+    name: LocalizedStringInput
+    group: String!
+    level: Int!
```

Comment:
I don't think the group and level of classifications should be subject to updates. They're kind of the primary key. If you want to change them, its actually just time to create a new Classification.

Reply:
I generally would agree with the **group** value not changing (although, there is talk, for example, of [CS becoming IT](https://pipsc.ca/groups/cs/cs-group-conversion-to-it-group)), but the **level** could change from one collective bargaining agreement to the next with the addition or subtraction of steps. I will remove both though if you still feel the same way after reading these comments.

Reply:
The CS to IT transition is a really good point - Okay, lets allow for renaming of group. Level I still don't see a reason to update.